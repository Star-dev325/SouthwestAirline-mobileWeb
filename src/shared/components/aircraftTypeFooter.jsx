// @flow
import React from 'react';
import cx from 'classnames';
import i18n from '@swa-ui/locale';

type Props = {
  dark?: boolean
};

const AircraftTypeFooter = (props: Props) => {
  const { dark } = props;

  return (
    <div
      className={cx('aircraft-type-footer', {
        'aircraft-type-footer': !dark,
        'aircraft-footer-dark': dark
      })}
    >
      <span>{i18n('SHARED__AIRCRAFT_TYPE_FOOTER__DESCRIPTION')}</span>
    </div>
  );
};

export default AircraftTypeFooter;
