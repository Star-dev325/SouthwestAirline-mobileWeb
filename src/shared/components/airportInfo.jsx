// @flow
import React from 'react';
import _ from 'lodash';

type Props = {
  airportInfo: {
    airportCode: string,
    airportName?: string,
    cityState?: string,
    country?: ?string
  },
  showDetail?: boolean
};

const AirportInfo = ({ airportInfo, showDetail }: Props) => (
  <div className="airport-info">
    <div className="airport-info--code">{airportInfo.airportCode}</div>
    {showDetail && !_.isEmpty(airportInfo) && (
      <div className="airport-info--detail">
        {airportInfo.airportName}, {airportInfo.cityState || airportInfo.country}
      </div>
    )}
  </div>
);

export default AirportInfo;
