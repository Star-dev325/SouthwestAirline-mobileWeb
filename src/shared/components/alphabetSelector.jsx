// @flow
import _ from 'lodash';
import React from 'react';
import ReactDOM from 'react-dom';
import withBodyClass from 'src/shared/enhancers/withBodyClass';
import type { ElementRef, HeaderRefsType } from 'src/shared/flow-typed/shared.types';

type Props = {
  shouldShow: ?boolean,
  alphabet: Array<string>,
  scrollTo: (string) => void
};

const AlphabetSelector = (props: Props) => {
  const letterRefs: HeaderRefsType = {};
  const { alphabet, shouldShow, scrollTo } = props;

  const _preventDefault = (event: Event) => event && event.preventDefault();

  const _onClick = (letter: string) => (event: Event) => {
    _preventDefault(event);
    scrollTo(letter);
  };

  const _onTouchMove = (event: Event) => {
    const touchLocation = _.chain(event).get('touches').head().get('clientY').value();
    const letterLocations = _.mapValues(letterRefs, (ref) => _.get(ref.getBoundingClientRect(), 'top'));

    const lettersAboveTouch = _.pickBy(letterLocations, (letterLocation) => letterLocation < touchLocation);
    const letterTouched = _.chain(lettersAboveTouch).keys().max().value();

    letterTouched && scrollTo(letterTouched);
  };

  const _setLetterRef = (letter: string) => (ref: ElementRef) => {
    _.set(letterRefs, letter, ReactDOM.findDOMNode(ref));
  };

  const _displayLetter = (letter: string) => (
    <div onClick={_onClick(letter)} ref={_setLetterRef(letter)} key={letter} data-qa={'alpha-select-letter'}>
      {letter}
    </div>
  );

  return (
    <div className="alphabet-selector disable-scrolling" onTouchMove={_onTouchMove}>
      {shouldShow && _.map(_.sortBy(alphabet), _displayLetter)}
    </div>
  );
};

export default withBodyClass('overflow-hidden')(AlphabetSelector);
