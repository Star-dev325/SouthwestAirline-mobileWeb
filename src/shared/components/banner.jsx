// @flow

import React from 'react';
import _ from 'lodash';
import cx from 'classnames';
import { flightStatusCssClassMapping } from 'src/flightStatus/constants/flightStatusCssClassMapping';

type Props = {
  text: string,
  className?: string,
  type?: string,
  boardingTime?: string
};

const Banner = ({ text, className, type, boardingTime }: Props) => {
  const bannerTypeStyle = `banner_${flightStatusCssClassMapping[type]}`;

  return (
    <div
      className={cx('banner', {
        [_.toString(bannerTypeStyle)]: !!type,
        [_.toString(className)]: !!className
      })}
    >
      <div className="banner-name">{text}</div>
      {boardingTime && <div className="boarding-time">{boardingTime}</div>}
    </div>
  );
};

export default Banner;
