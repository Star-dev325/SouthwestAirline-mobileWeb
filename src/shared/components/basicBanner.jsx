// @flow
import cx from 'classnames';
import React from 'react';
import Icon from 'src/shared/components/icon';
import ICON_CLASSES from 'src/shared/constants/iconConstants';

type Props = {
  className?: string,
  icon?: string,
  iconClassName?: string,
  message?: string,
  subtitleClassName?: string,
  title?: string
};

const BasicBanner = (props: Props) => {
  const { className, icon, iconClassName, message, subtitleClassName, title } = props;

  return (
    <div className={cx('basic-banner', className)}>
      {icon && (
        <div>
          <Icon className={iconClassName} type={ICON_CLASSES[icon] || ICON_CLASSES.WARNING} />
        </div>
      )}
      <div className="banner-container">
        <b className="banner-container--title">{title}</b>
        <em className={cx('banner-container--subtitle', subtitleClassName)}>{message}</em>
      </div>
    </div>
  );
};

export default BasicBanner;
