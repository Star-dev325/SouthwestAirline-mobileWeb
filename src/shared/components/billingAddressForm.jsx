// @flow
import React from 'react';
import Segments from 'src/shared/components/segments';
import Segment from 'src/shared/components/segment';
import PageHeaderWithButtons from 'src/shared/components/pageHeaderWithButtons';
import i18n from '@swa-ui/locale';
import Form from 'src/shared/form/components/form';
import withForm from 'src/shared/form/enhancers/withForm';
import ContactInfoFields from 'src/shared/form/fields/contactInfoFields';
import billingAddressFormValidator from 'src/shared/form/formValidators/billingAddressFormValidator';

import type { FormData } from 'src/shared/form/flow-typed/form.types';

type Props = {
  formId: string,
  initialFormData?: FormData,
  goBack: () => void,
  onSubmit: () => void
};

class BillingAddressForm extends React.Component<Props> {
  render() {
    const { formId, goBack, onSubmit } = this.props;

    return (
      <Form formId={formId} onSubmit={onSubmit} className="billing-info-form">
        <PageHeaderWithButtons
          title={i18n('SHARED__ADD__BILLING__ADDRESS')}
          leftButtons={[{ name: i18n('SHARED__BUTTON_TEXT__CANCEL'), onClick: goBack }]}
          rightButtons={[{ name: i18n('SHARED__BUTTON_TEXT__DONE'), type: 'submit' }]}
        />
        <Segments>
          <Segment>
            <ContactInfoFields
              names={[
                'isoCountryCode',
                'addressLine1',
                'addressLine2',
                'city',
                'stateProvinceRegion',
                'zipOrPostalCode',
                'phoneNumber',
                'phoneCountryCode'
              ]}
              label={i18n('SHARED__BILLING_INFO__BILLING_ADDRESS')}
              supportModifyCountryCode
            />
          </Segment>
        </Segments>
      </Form>
    );
  }
}

export default withForm({
  autoClearFormData: true,
  formValidator: billingAddressFormValidator,
  defaultValues: (props: Props) => {
    const { initialFormData } = props;

    if (initialFormData) return initialFormData;

    return {
      isoCountryCode: 'US',
      phoneCountryCode: 'US'
    };
  }
})(BillingAddressForm);
