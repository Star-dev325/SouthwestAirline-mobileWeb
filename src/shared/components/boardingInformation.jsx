// @flow
import React from 'react';
import i18n from '@swa-ui/locale';

type Props = {
  boardingGate?: string,
  boardingGroup: ?string,
  boardingPosition: ?string
};

const BoardingInformation = (props: Props) => {
  const renderWithBoardingGate = () => {
    const { boardingGate, boardingGroup, boardingPosition } = props;

    return (
      <div className="boarding-information--item-row">
        <div className="boarding-information--item">
          <div className="boarding-information--item-label">{i18n('SHARED__BOARDING_INFORMATION__GATE')}</div>
          <div className="boarding-information--item-info">{boardingGate}</div>
        </div>
        <div className="boarding-information--item">
          <div className="boarding-information--item-label">{i18n('SHARED__BOARDING_INFORMATION__GROUP')}</div>
          <div className="boarding-information--item-info">{boardingGroup}</div>
        </div>
        <div className="boarding-information--item">
          <div className="boarding-information--item-label">{i18n('SHARED__BOARDING_INFORMATION__POSITION')}</div>
          <div className="boarding-information--item-info_right">{boardingPosition}</div>
        </div>
      </div>
    );
  };

  const renderWithoutBoardingGate = () => {
    const { boardingGroup, boardingPosition } = props;

    return (
      <div className="boarding-information--without-gate">
        <span>
          <span className="without-gate--item-label">{i18n('SHARED__BOARDING_INFORMATION__BOARDING_GROUP')}</span>
          <span className="without-gate--item-content"> {boardingGroup} </span>
        </span>
        <span className="without-gate--item-row">
          <span className="without-gate--item-label">{i18n('SHARED__BOARDING_INFORMATION__POSITION')}</span>
          <span className="without-gate--item-content"> {boardingPosition} </span>
        </span>
      </div>
    );
  };

  let content = renderWithoutBoardingGate();

  if (props.boardingGate) {
    content = renderWithBoardingGate();
  }

  return <div className="boarding-info boarding-information center pt5 pb5 mt3">{content}</div>;
};

export default BoardingInformation;
