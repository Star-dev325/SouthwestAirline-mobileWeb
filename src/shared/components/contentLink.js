// @flow
import React from 'react';

import type { Node } from 'react';
import { connect } from 'react-redux';
import _ from 'lodash';

type Props = {
  className?: string,
  href?: ?string,
  id?: string,
  dataQa?: string,
  children?: Node,
  shouldOpenLinkInSelf?: boolean,
  raw?: ?string,
  hidden?: boolean,
  isWebView?: ?boolean,
  target?: string,
  onClick?: (*) => void
};

export const ContentLink = ({
  className,
  raw,
  onClick,
  href,
  id,
  children,
  dataQa,
  hidden,
  isWebView,
  target,
  shouldOpenLinkInSelf = true
}: Props) => {
  let component = null;

  const handleButtonClick = (event: Event) => {
    event.preventDefault();
    event.stopPropagation();
    onClick && onClick(event);
  };

  if (!hidden) {
    if (raw) {
      const finalHtml = isWebView && shouldOpenLinkInSelf ? raw.replace('_blank', '_self') : raw;

      component = (
        <span className={className} id={id} data-qa={dataQa} dangerouslySetInnerHTML={{ __html: finalHtml }} />
      );
    } else if (href && children) {
      const derivedTarget = isWebView ? '_self' : '_blank';

      component = (
        <a className={className} id={id} data-qa={dataQa} href={href} target={target || derivedTarget}>
          {children}
        </a>
      );
    } else if (onClick) {
      component = (
        <button className={`button--link ${className ? className : ''}`} id={id} onClick={handleButtonClick}>
          {children}
        </button>
      );
    } else {
      component = (
        <p className={className} id={id} data-qa={dataQa}>
          {children}
        </p>
      );
    }
  }

  return component;
};

const mapStateToProps = (state) => ({
  isWebView: _.get(state, 'app.webView.isWebView')
});

export default connect(mapStateToProps, {})(ContentLink);
