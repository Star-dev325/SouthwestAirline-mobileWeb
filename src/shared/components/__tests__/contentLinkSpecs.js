import { ContentLink } from 'src/shared/components/contentLink';
import { shallow, mount } from 'enzyme';
import React from 'react';
import _ from 'lodash';

const fareRulesWithLinks =
  'Please read <a href="https://mobile.southwest.com/fare-rules" target="_blank">fare rules</a> associated with purchase.';
const baseProps = {
  className: 'mock-class',
  id: 'mock-id',
  dataQa: 'mock-data-qa'
};

describe('ContentLink', () => {
  context('default', () => {
    it('should render with href', () => {
      const comp = shallow(
        <ContentLink {...baseProps} href="/link">
          This is a link
        </ContentLink>
      );

      expect(comp).toMatchSnapshot();
    });

    it('should render with provided target', () => {
      const comp = shallow(
        <ContentLink {...baseProps} href="/link" target="_self">
          This is a link
        </ContentLink>
      );

      expect(comp).toMatchSnapshot();
    });

    it('should render with raw html', () => {
      const comp = mount(<ContentLink raw={fareRulesWithLinks}>This is a link</ContentLink>);

      expect(comp.text()).to.equal('Please read fare rules associated with purchase.');
      expect(comp.html()).to.contain('https://mobile.southwest.com/fare-rules');
      expect(comp.html()).to.contain('_blank');
    });

    it('should not render anchor when hidden', () => {
      const comp = shallow(
        <ContentLink href="/link" hidden>
          This is a link
        </ContentLink>
      );

      expect(comp.text()).to.equal('');
      expect(comp.find('a')).not.to.exist;
    });

    it('should render only text when missing required href', () => {
      const comp = shallow(<ContentLink {...baseProps}>This is a link</ContentLink>);

      expect(comp).toMatchSnapshot();
    });

    it('should render button with onClick', () => {
      const comp = shallow(
        <ContentLink {...baseProps} onClick={_.noop}>
          This is a link
        </ContentLink>
      );

      expect(comp).toMatchSnapshot();
    });

    it('should not render anchor when missing required props', () => {
      const comp = shallow(<ContentLink />);

      expect(comp.text()).to.equal('');
      expect(comp.find('a')).not.to.exist;
    });
  });

  context('with web view', () => {
    it('should render with href', () => {
      const comp = shallow(
        <ContentLink {...baseProps} href="/link" isWebView>
          This is a link
        </ContentLink>
      );

      expect(comp).toMatchSnapshot();
    });

    it('should render with raw and open in new tab', () => {
      const comp = shallow(
        <ContentLink raw={fareRulesWithLinks} isWebView shouldOpenLinkInSelf={false}>
          This is a link
        </ContentLink>
      );

      expect(comp).toMatchSnapshot();
    });

    it('should render with raw html', () => {
      const comp = mount(
        <ContentLink raw={fareRulesWithLinks} isWebView>
          This is a link
        </ContentLink>
      );

      expect(comp.text()).to.equal('Please read fare rules associated with purchase.');
      expect(comp.html()).to.contain('https://mobile.southwest.com/fare-rules');
      expect(comp.html()).to.contain('_self');
    });

    it('should not render anchor when hidden', () => {
      const comp = shallow(
        <ContentLink {...baseProps} href="/link" hidden isWebView>
          This is a link
        </ContentLink>
      );

      expect(comp.text()).to.equal('');
      expect(comp.find('a')).not.to.exist;
    });

    it('should render only text when missing required href', () => {
      const comp = shallow(
        <ContentLink {...baseProps} isWebView>
          This is a link
        </ContentLink>
      );

      expect(comp).toMatchSnapshot();
    });
  });
});
