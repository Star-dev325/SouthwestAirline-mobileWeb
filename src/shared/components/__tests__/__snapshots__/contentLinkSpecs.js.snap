// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentLink default should render button with onClick 1`] = `
<button
  className="button--link mock-class"
  id="mock-id"
  onClick={[Function]}
>
  This is a link
</button>
`;

exports[`ContentLink default should render only text when missing required href 1`] = `
<p
  className="mock-class"
  data-qa="mock-data-qa"
  id="mock-id"
>
  This is a link
</p>
`;

exports[`ContentLink default should render with href 1`] = `
<a
  className="mock-class"
  data-qa="mock-data-qa"
  href="/link"
  id="mock-id"
  target="_blank"
>
  This is a link
</a>
`;

exports[`ContentLink default should render with provided target 1`] = `
<a
  className="mock-class"
  data-qa="mock-data-qa"
  href="/link"
  id="mock-id"
  target="_self"
>
  This is a link
</a>
`;

exports[`ContentLink with web view should render only text when missing required href 1`] = `
<p
  className="mock-class"
  data-qa="mock-data-qa"
  id="mock-id"
>
  This is a link
</p>
`;

exports[`ContentLink with web view should render with href 1`] = `
<a
  className="mock-class"
  data-qa="mock-data-qa"
  href="/link"
  id="mock-id"
  target="_self"
>
  This is a link
</a>
`;

exports[`ContentLink with web view should render with raw and open in new tab 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "Please read <a href=\\"https://mobile.southwest.com/fare-rules\\" target=\\"_blank\\">fare rules</a> associated with purchase.",
    }
  }
/>
`;
